### ISSUE TITLE
Weird training result

### ISSUE THREAD

**Issue by dongyunkim-arch** [eladrich/pixel2style2pixel#216]
Hello! 
Thanks for this amazing model and well documented description.

I am trying to train my own dataset(buildings), but trained results are pretty weird, especially weird pattern covers the whole part.
This pattern wasn't shown in my generated images by stylegan model, but it doesn't disappear in 25000epoch. 

![23800](https://user-images.githubusercontent.com/70664694/140630160-fbbd93ec-0569-45fd-af50-fff0bd94fabd.jpg)
![23200](https://user-images.githubusercontent.com/70664694/140630164-3b3a4797-dffe-4f8c-a6d5-d1f177f560f3.jpg)

This is my training dataset and random images generated by stylegan.

![tr](https://user-images.githubusercontent.com/70664694/140630216-e5d657f6-873a-4463-8409-c550a5f22e57.JPG)

![Capture2](https://user-images.githubusercontent.com/70664694/140630226-eb13e687-b43e-4427-8339-e06f38705688.JPG)

and this is my training options.

--workers=8 --batch_size=8 
--test_batch_size=8 --test_workers=8 
--val_interval=2500 --save_interval=5000 
--encoder_type=GradualStyleEncoder 
--start_from_latent_avg 
--lpips_lambda=0.8 
--l2_lambda=1 
--id_lambda=0.1

Do you have any insight or thought what's wrong here....?

**Comment by yuval-alaluf** [comment]
Everything looks good in terms of inputs and models. I've seen these artifacts before, mainly when working on harder domains like yours here. To help with this, I found the MoCo-basrd similarity loss can help (it is similar to the ID similarity loss for faces). 
If the MoCo loss doesn't improve results, you could look into working with ReStyle which builds on pSp and showed much better results on more challenging domains.

**Comment by dongyunkim-arch** [comment]
Thank you for the advice!
From your advice, I changed the loss to MoCo loss, and the result is getting better compared to previous one I guess.

I would like to ask you regarding my result. I trained it around half a day and got this result (I am using rtx 3090 as eGPU). It seems that the model captures the overall shape but when I goes into detail of building, it doesn't look great, lost lots of detail.... So, do you think this is due to short training time or there could be some problems such as dataset or domain?
If you advice some future direction, that would be so helpful!
![27500](https://user-images.githubusercontent.com/70664694/140846135-6340de66-3ce3-4428-aa4e-352da0f5e800.jpg)
![27400](https://user-images.githubusercontent.com/70664694/140846143-20db36d8-3330-414c-a6b0-d1563afa4d82.jpg)
![27300](https://user-images.githubusercontent.com/70664694/140846145-43c70794-aec2-4910-ab8d-dcbd8a0b3853.jpg)

Thank you!

**Comment by yuval-alaluf** [comment]
It could also be that pSp is simply not strong enough to invert and reconstruct these challenging images accurately enough.  If you still don't see an improvement, you could check out ReStyle, which showed much better results on non-facial domains. Hopefully, you'll be able to get better reconstructions there.
